src/libraries/System.Text.Json/tests/Serialization/DynamicTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class DynamicTests
+    {
+        public const string Json =
+            "{\"MyString\":\"Hello\",\"MyNull\":null,\"MyBoolean\":true,\"MyArray\":[1,2],\"MyInt\":42,\"MyDouble\":4.2,\"MyDateTime\":\"2020-07-08T00:00:00\",\"MyGuid\":\"ed957609-cdfe-412f-88c1-02daca1b4f51\",\"MyObject\":{\"MyString\":\"World\"}}";
+        public static DateTime MyDateTime => new DateTime(2020, 7, 8);
+        public static Guid MyGuid => new Guid("ed957609-cdfe-412f-88c1-02daca1b4f51");
+
+        internal static ExpandoObject GetExpandoObject()
+        {
+            dynamic myDynamicChild = new ExpandoObject();
+            myDynamicChild.MyString = "World";
+
+            dynamic myDynamic = new ExpandoObject();
+            myDynamic.MyString = "Hello";
+            myDynamic.MyNull = null;
+            myDynamic.MyBoolean = true;
+            myDynamic.MyArray = new List<int>() { 1, 2 };
+            myDynamic.MyInt = 42;
+            myDynamic.MyDouble = 4.2;
+            myDynamic.MyDateTime = MyDateTime;
+            myDynamic.MyGuid = MyGuid;
+            myDynamic.MyObject = myDynamicChild;
+
+            // Verify basic dynamic support.
+            int c = myDynamic.MyInt;
+            Assert.Equal(42, c);
+
+            return myDynamic;
+        }
+
+        [Fact]
+        public static void DynamicKeyword()
+        {
+            dynamic myDynamic = GetExpandoObject();
+
+            // STJ can serialize with ExpandoObject + 'dynamic' keyword.
+            string json = JsonSerializer.Serialize<dynamic>(myDynamic);
+            Assert.Equal(Json, json);
```

Comment:
`dynamic` is different since internally it uses a dictionary to serialize and not based on reflection order. However, dictionary enumeration through KeyValuePair is not actually considered deterministic although in this case it will be (with current implementation of Dictionary anyway). In either case, AssertJsonEqual is a good idea and I'll change to that.